File: planik-android/app/src/main/java/pl/planik/local/entity/PlanDayEntryEntity.kt
package pl.planik.local.entity

import androidx.room.*
import java.time.OffsetDateTime
import java.time.ZoneId

@Entity(
  tableName = "plan_day_entries",
  foreignKeys = [
    ForeignKey(
      entity = UserEntity::class,
      parentColumns = ["id"],
      childColumns = ["plan_id"],
      onDelete = ForeignKey.CASCADE,
      onUpdate = ForeignKey.CASCADE
    )
  ],
  indices = [
    Index(value = ["plan_id"])
  ]
)
data class PlanDayEntryEntity(
  @ColumnInfo(name = "plan_id") val planId: Int,
  @ColumnInfo(name = "day_of_week") val dayOfWeek: Int,
  @ColumnInfo(name = "title") val title: String,
  @ColumnInfo(name = "start") val start: OffsetDateTime,
  @ColumnInfo(name = "end") val end: OffsetDateTime,
  @ColumnInfo(name = "created_at") val createdAt: OffsetDateTime? = OffsetDateTime.now(ZoneId.of("UTC")),
  @ColumnInfo(name = "updated_at") val updatedAt: OffsetDateTime? = null
) {
  @ColumnInfo(name = "id")
  @PrimaryKey(autoGenerate = true)
  var id: Int = 0
}

File: planik-flutter/planik/lib/local/entity/plan_day_entry_entity.dart
import 'package:floor/floor.dart';
import 'package:flutter/material.dart';
import 'package:planik/local/entity/plan_entity.dart';

@Entity(
  tableName: 'plan_day_entries',
  foreignKeys: [
    ForeignKey(
      childColumns: ['plan_id'],
      parentColumns: ['id'],
      entity: PlanEntity,
      onDelete: ForeignKeyAction.cascade,
    )
  ],
)
@immutable
class PlanDayEntryEntity {
  const PlanDayEntryEntity({
    required this.id,
    required this.planId,
    required this.dayOfWeek,
    required this.name,
    required this.start,
    required this.end,
    required this.createdAt,
    required this.updatedAt,
  });

  @PrimaryKey(autoGenerate: true)
  final int? id;
  @ColumnInfo(name: 'plan_id')
  final int planId;
  @ColumnInfo(name: 'day_of_week')
  final int dayOfWeek;
  final String name;
  final String start;
  final String end;
  @ColumnInfo(name: 'created_at')
  final String createdAt;
  @ColumnInfo(name: 'updated_at')
  final String? updatedAt;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is PlanDayEntryEntity &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          planId == other.planId &&
          dayOfWeek == other.dayOfWeek &&
          name == other.name &&
          start == other.start &&
          end == other.end &&
          createdAt == other.createdAt &&
          updatedAt == other.updatedAt;

  @override
  int get hashCode =>
      id.hashCode ^
      planId.hashCode ^
      dayOfWeek.hashCode ^
      name.hashCode ^
      start.hashCode ^
      end.hashCode ^
      createdAt.hashCode ^
      updatedAt.hashCode;

  PlanDayEntryEntity copyWith({
    int? id,
    int? planId,
    int? dayOfWeek,
    String? name,
    String? start,
    String? end,
    String? createdAt,
    String? updatedAt,
  }) {
    return PlanDayEntryEntity(
      id: id ?? this.id,
      planId: planId ?? this.planId,
      dayOfWeek: dayOfWeek ?? this.dayOfWeek,
      name: name ?? this.name,
      start: start ?? this.start,
      end: end ?? this.end,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
